### ISSUE TITLE
Inconsistent and mostly undocumented default exit codes

### ISSUE THREAD

**Issue by haasn** [google/sanitizers#1225]
Turns out my CI was completely ignoring msan failures for the longest time because an exit code of 77 (msan's default `exitcode`) is interpreted as "skip this test" in the GNU testing protocol (used by meson etc.).

I found it essentially entirely impossible to figure out this fact from the documentation alone, I was only able to find [SanitizerCommonFlags](https://github.com/google/sanitizers/wiki/SanitizerCommonFlags) which claims that `exitcode` defaults to 1, as well as various other bits of documentation that don't even mention the exit code.

But looking at the source code of rt-lib, the actual behavior of the exit code seems a bit sporadic:

- msan defaults to `77` (`msan/msan.cpp`)
- asan defaults to `1` (`asan/asan_flags.cpp`)
- ubsan is hard-coded to `Die()` (which implies `exit(1)`), and ignores `cf.exitcode` entirely!
- tsan defaults to `66` (`tsan/rtl/tsan_flags.cpp`)
- lsan defaults to `23` (`lsan/lsan.cpp`)
- hwasan defaults to `99` (`hwasan/hwasan.cpp`)
- scudo defaults to `1` (`scudo/scudo_flags.cpp`)
- ...

Quite frankly this needs to be documented more clearly at least *somewhere*, so that users aren't left grepping the source code to figure out what tools exit with what status codes and how to change them.

As a side note, exiting with `77` definitely should be changed for the reason mentioned, although rather than setting it to something else absurd like `55`, I think the behavior and default should be consistent across all of these tools.

**Comment by yugr** [comment]
+1 for having tools abort instead of exit.

**Comment by haasn** [comment]
Oh, yes, `abort()` is also a good idea, specifically because it implies the generation of core dumps which can assist in debugging the issues involved.

**Comment by eugenis** [comment]
Yes, we are not being very consistent here. 77 was picked at random (and it
also looks nice).
I'm not sure about changing the defaults at this point, there could be
users who depend on these values.

**Comment by Flowdalic** [comment]
Related change request https://reviews.llvm.org/D92490
